esm_runscripts/namelists.py

```diff
@@ -126,7 +126,33 @@ def nmls_remove(mconfig):
                     value = change_entries[key]
                     if value == "remove_from_namelist":
                         namelist_removes.append((namelist, change_chapter, key))
+
+                        # the key is probably coming from esm_tools config
+                        # files or from a user runscript. It can contain lower
+                        # case, but the original Fortran namelist could be in
+                        # any case combination. Here `original_key` is coming
+                        # from the default namelist and may contain mixed case.
+                        # `key` is the processed variable my f90nml package and
+                        # is lowercase
+                        remove_original_key = False
+
+                        # traverse the namelist chapter and see if a mixed case
+                        # variable is also found
+                        for key2 in namelist_changes[namelist][change_chapter]:
+                            # take care of the MiXeD FORTRAN CaSeS
+                            if key2.lower() == key.lower() and key2 != key:
```

Comment:
That's needed because Python dicts support upper/lower case keys, but FORTRAN handles it all the same, right?

Reply:
Inside the chapter there can be 2 different versions of a variable. One is what the default Fortran namelist contains (eg. `TideForcingPath`), and the other one is what the user provides in the runscript (eg., `tideforcingpath`). What I understood from debugging is that f90nml converts everything to lowercase. Here is the default namelist (inside `esm_tools`)

```fortran
&paths
MeshPath=<meshpath>
OpbndPath=''
ClimateDataPath=<climatedatapath>
ForcingDataPath=<forcingdatapath>
TideForcingPath=<tideforcingpath>
ResultPath=<resultpath>
/
```

And this is the namelist in the `workdir`: 

```fortran
&paths
    meshpath = '/work/ollie/pool/FESOM/fesom2.0/meshes/mesh_CORE2_final/'
    opbndpath = 'somepath'
    climatedatapath = '/work/ollie/pool/FESOM//hydrography/'
    forcingdatapath = '/work/ollie/pool/FESOM//forcing/'
    tideforcingpath = 'somepath'
    resultpath = '/work/ollie/dural/claudia//my_first_test1//run_20010101-20011231/work/'
/
```